common/lib/client/auth.js

```diff
@@ -76,35 +74,17 @@ var Auth = (function() {
 			Logger.logAction(Logger.LOG_ERROR, 'Auth()', msg);
 			throw new Error(msg);
 		}
-		/* using token auth, but decide the method */
-		this.method = 'token';
-		if(options.token) {
-			/* options.token may contain a token string or, for convenience, a TokenDetails */
-			options.tokenDetails = (typeof(options.token) === 'string') ? {token: options.token} : options.token;
-		}
-		this.tokenDetails = options.tokenDetails;
-
-		if(options.authCallback) {
-			Logger.logAction(Logger.LOG_MINOR, 'Auth()', 'using token auth with authCallback');
-		} else if(options.authUrl) {
-			Logger.logAction(Logger.LOG_MINOR, 'Auth()', 'using token auth with authUrl');
-		} else if(options.keySecret) {
-			Logger.logAction(Logger.LOG_MINOR, 'Auth()', 'using token auth with client-side signing');
-		} else if(options.tokenDetails) {
-			Logger.logAction(Logger.LOG_MINOR, 'Auth()', 'using token auth with supplied token only');
-		} else {
-			var msg = 'options must include valid authentication parameters';
-			Logger.logAction(Logger.LOG_ERROR, 'Auth()', msg);
-			throw new Error(msg);
-		}
+
+		/* using token auth */
+		this._saveTokenOptions(options.defaultTokenParams, options);
+		this._logTokenAuthMethod();
```

Comment:
Given this is log & validate valid authentication params, it should have an appropriate name.